File: Math/871/divisor.cpp
#include <iostream>
#include <utility>
#include <unordered_map>
using namespace std;

const int MOD = 1e9 + 7;
int n, t;
long long res = 1, p, s;
unordered_map<int, int> primes;

void find_primes(int n);

int main() {
    scanf("%d", &n);
    while (n--) {
        scanf("%d", &t);
        find_primes(t);
    }
    for (pair<int, int> prime: primes) {
        p = prime.first, s = 1;
        for (int i = 0; i < prime.second; i++) s = (s * p + 1) % MOD;
        res = (res * s) % MOD;
    }
    printf("%lld", res);
    return 0;
}

void find_primes(int n) {
    for (int i = 2; i <= n / i; i++) {
        while (!(n % i)) primes[i]++, n /= i;
    }
    if (n > 1) primes[n]++;
}

File: Math/871/divisor.py
from collections import defaultdict

if __name__ == "__main__":
    n = int(input())
    MOD = 10 ** 9 + 7
    primes = defaultdict(int)
    for _ in range(n):
        a = int(input())
        i = 2
        while i <= a // i:
            if a % i == 0:
                while a % i == 0:
                    primes[i] += 1
                    a //= i
            i += 1
        if a > 1:
            primes[a] += 1
    res = 1
    for prime, cnt in primes.items():
        t = 1
        for _ in range(cnt):
            t = (t * prime + 1) % MOD
        res = res * t % MOD
    print(res)
